src/main/scala/is/hail/methods/SampleQC.scala

```diff
@@ -69,55 +69,59 @@ class SampleQCCombiner(val keepStar: Boolean) extends Serializable {
 
   // FIXME per-genotype
 
-  def merge(v: Variant, ACs: Array[Int], g: Genotype): SampleQCCombiner = {
+  def merge(v: Variant, acs: Array[Int], g: Genotype): SampleQCCombiner = {
 
-    g.gt match {
+    val gt = g.unboxedGT
+    if (gt < 0)
+      nNotCalled += 1
+    else {
 
-      case Some(0) =>
+      // FIXME these should probably get merged if gt is uncalled,
+      // but don't want to break behavior without a version change
+      if (g.unboxedDP >= 0)
+        dpSC.merge(g.unboxedDP)
+
+      if (g.unboxedGQ >= 0)
+        gqSC.merge(g.unboxedGQ)
+
+      if (gt == 0) {
         nHomRef += 1
+      } else {
+        val gtPair = Genotype.gtPair(gt)
 
-      case Some(gt) =>
-        val nonRefAlleleIndices = Genotype.gtPair(gt).alleleIndices.filter(i => i > 0 && (keepStar || !v.altAlleles(i - 1).isStar))
+        def mergeAllele(ai: Int) {
+          if (ai > 0 && (keepStar || !v.altAlleles(ai - 1).isStar)) {
 
-        if (!nonRefAlleleIndices.isEmpty) {
-          nonRefAlleleIndices.foreach({
-            ai =>
-              val altAllele = v.altAlleles(ai - 1)
-              if (altAllele.isSNP) {
+            val altAllele = v.altAlleles(ai - 1)
+            altAllele.altAlleleType match {
+              case AltAlleleType.SNP =>
                 nSNP += 1
                 if (altAllele.isTransition)
                   nTi += 1
-                else {
-                  assert(altAllele.isTransversion)
+                else
                   nTv += 1
-                }
-              } else if (altAllele.isInsertion)
+              case AltAlleleType.Insertion =>
                 nIns += 1
-              else if (altAllele.isDeletion)
+              case AltAlleleType.Deletion =>
                 nDel += 1
+              case _ =>
+            }
 
-              if (ACs(ai - 1) == 1)
-                nSingleton += 1
-          })
-
-          if (nonRefAlleleIndices.length == 1 || nonRefAlleleIndices(0) != nonRefAlleleIndices(1))
-            nHet += 1
-          else
-            nHomVar += 1
+            if (acs(ai) == 1)
+              nSingleton += 1
+          }
         }
-      case None =>
-        nNotCalled += 1
 
-    }
+        mergeAllele(gtPair.j)
+        mergeAllele(gtPair.k)
 
-    if (g.isCalled) {
-      g.dp.foreach { v =>
-        dpSC.merge(v)
-      }
-      g.gq.foreach { v =>
-        gqSC.merge(v)
+        if (gtPair.j == gtPair.k)
```

Comment:
We should have handwritten tests for all parameters computed by sampleQC and variantQC for a small example. This should have been caught by a test. I see some sample QC output is tested in `FilterSuite`, but only `nCalled` and `dpMean`.